M/MLX/bundled/patches/cmake-warning-as-error.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d3df54878..e716e925a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -81,7 +81,6 @@ include(FetchContent)
+ cmake_policy(SET CMP0135 NEW)
+ 
+ add_library(mlx)
+-set_target_properties(mlx PROPERTIES COMPILE_WARNING_AS_ERROR ON)
```

Comment:
According to this https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_WARNING_AS_ERROR.html, we can instead pass `--compile-no-warning-as-error ` on the CMake command line to ignore this property, so then we won't need to patch it away.